### DISCUSSION TITLE
How to get a primary outline button

### DISCUSSION THREAD

**Discussion by seyaobey** [chakra-ui/chakra-ui#3351] [Q&A] [answered]
I've been using chakra-ui for a few weeks now and I'm loving it. I've got a request from my designer to get a button with a list of basic colors (`primary, info, warning, danger`) with the `outline` option (boolean). I'm wondering how I can accomplish this with chakra-ui. At the moment there is already a `outline variant`.

These are the combinations I would like to achieve (assuming I'm using the `colorScheme` prop to select color):

`<Button colorScheme="primary" size="..." />`
`<Button colorScheme="primary" outline={true} />`

(same goes for `info, warning, danger` colors)

Here is a visual example of how this could look like (taken from bootstrap inspired template):
`<Button colorScheme="primary"  />`
![image](https://user-images.githubusercontent.com/2738569/107564306-869ab980-6be2-11eb-84c9-0c847e72edaa.png)

`<Button colorScheme="primary" outline />`
![image](https://user-images.githubusercontent.com/2738569/107565085-85b65780-6be3-11eb-9b8f-6f7f0911e75b.png)

This is what I've been trying to do so far:
1. I've created a new color in my theme
```
const theme = extendTheme({
...
   colors: {
       primary: 'xxx'
   }
})
```
2. I apply this theme in my button:
```
<Button colorScheme="primary" variant="ouline" {...}/>
```

This is the result (good!§):
![image](https://user-images.githubusercontent.com/2738569/107565625-23aa2200-6be4-11eb-9170-44dd260ebcbb.png)

Now I'm trying to get the same effect like in bootstrap whereby when you hover over an `outline button`, then the background and color revert to how they are in `solid` mode (in other words on hover, background would be `primary color` and color would be `white`). 

How can I accomplish this in a "chakra-esque fashion"? This is how I'm trying to modify my theme object to achieve it:

```
const theme = extendTheme({
...
   colors: {
       primary: 'xxx'
   },
  components: {
       Button: {
            variants: {
                outline: {
                   _hover: {} // <-- what shoud I write here so that on hover, I apply in different background whether it is primary,
                                    // info, warning or danger
                }
            }
       } 
  }
})
```

I'm not even sure whether my approach with the custom theme is the right way to go.
If there is no solution to achieve this behaviour with `chakra-ui`, I'm thinking of using `styled-componnet` for this particular case.

Thanks

**Comment by chr-ge** [comment, marked as answer]
You can get the `colorMode` prop and do something like this:
```jsx
const theme = extendTheme({
  components: {
    Button: {
      variants: {
        outline: (props) => ({
          _hover: {
            bg: props.colorScheme // option 1
            bg: `${props.colorScheme}.500` // option 2
          }
        })
      }
    }
  }
})
```

**Reply by seyaobey** [reply]
Wow, pretty cool! Thanks for the quick answer.
